harness/determined/searcher/core_search_runner.py

```diff
@@ -0,0 +1,121 @@
+import logging
```

Comment:
The file layout needs reconsideration to be suitable for external users and internal developers alike.

The file layout we use in newly added code is laid out in [this old rfc](https://github.com/rb-determined-ai/pedl/blob/9bd2168ec23fbba866194f3d7d3cfc243d267100/python_code_layout.md).

Basically, what should the user actually be importing in, and what should they be thinking about?

Should they need to think about `det.searcher`?  (my guess is probably)

Should they need to think about `det.searcher.core_search_runner`?  (my guess is probably not)

That means that this file probably needs to be named `_core_search_runner`, and `searcher/__init__.py` probably needs to expose some symbols from inside this file.